Acceptance Test for AWS instance failed when AWS Region is not us-east-1 in my AWS profile

Hello everyone!

**Description**

I tried to execute Acceptance Test for AWS instance with following command.

```
ACC_AWS_PROFILE=individual AWS_PROFILE=individual-read-only DRIFTCTL_ACC=true go test -run=TestAcc_ ./pkg/resource/aws/aws_instance_test.go
```

* `individual` is IAM User attached `AdministratorAccess` policy.
* `individual-read-only` is IAM User attached `ReadOnlyAccess` policy
* Region for above IAM User is defined as `ap-northeast-1` in `~/.aws/config`

But, above test have failed with following error.

```
...
--- FAIL: TestAcc_AwsInstance_WithBlockDevices (119.42s)
    aws_instance_test.go:59: Error, unexpected number of instances found, manual check required
FAIL
FAIL  command-line-arguments    120.458s
FAIL
```

**Environment**

 - OS: 
```
ProductName:	Mac OS X
ProductVersion:	10.15.2
```
 - driftctl version: Using main branch (HEAD = `78f1e40acc43e76dfd27173027870bc217c88e8e`)
- Go: 1.15.5 darwin/amd64

**How to reproduce**

Region for your IAM User is defined as `ap-northeast-1` in `~/.aws/config` (Precisely other than `us-east-1`)

**Possible Solution**

I investigated the reason this error occurs and finally found it. In following Terraform file, region is defined as `us-east-1`

https://github.com/cloudskiff/driftctl/blob/78f1e40acc43e76dfd27173027870bc217c88e8e/pkg/resource/aws/testdata/acc/aws_instance/providers.tf#L1-L3

And the environment variable `"AWS_REGION": "us-east-1"` is also defined in following file.

https://github.com/cloudskiff/driftctl/blob/78f1e40acc43e76dfd27173027870bc217c88e8e/pkg/resource/aws/aws_instance_test.go#L34-L41

On the other hand, the environment variable defined in `acceptance.AccCheck` is embedded as following file.

https://github.com/cloudskiff/driftctl/blob/78f1e40acc43e76dfd27173027870bc217c88e8e/test/acceptance/testing.go#L274-L283

It means `PreExec` is executed before the environment is embedded, so `aws ec2 describe-instances` in `PreExec` for Acceptance Test for AWS instance is executed specifying region as the one in `~/.aws/config`

https://github.com/cloudskiff/driftctl/blob/78f1e40acc43e76dfd27173027870bc217c88e8e/pkg/resource/aws/aws_instance_test.go#L39-L54

I think the environment variables should be embedded before `PreExec`, so above conditions should be swapped as following.

```
 if len(check.Env) > 0 { 
 	for key, value := range check.Env { 
 		os.Setenv(key, value) 
 	} 
 } 
 if check.PreExec != nil { 
 	c.useTerraformEnv() 
 	check.PreExec() 
 	c.restoreEnv() 
 } 
```

Actually I modified source and confirmed the test passed! This modification makes Acceptance Test more friendly to every developer all over the world!

If needed, I create PR about this issue :smiley:

**Additional context**
none.

Thanks a lot for reading my issue :smiley_cat:

--- a/test/acceptance/testing.go
+++ b/test/acceptance/testing.go
@@ -271,15 +271,15 @@ func Run(t *testing.T, c AccTestCase) {
 		if check.Check == nil {
 			t.Fatal("Check attribute must be defined")
 		}
-		if check.PreExec != nil {
-			c.useTerraformEnv()
-			check.PreExec()
-			c.restoreEnv()
-		}
 		if len(check.Env) > 0 {
 			for key, value := range check.Env {
 				os.Setenv(key, value)
 			}
+		}
+		if check.PreExec != nil {
+			c.useTerraformEnv()
+			check.PreExec()
+			c.restoreEnv()
 		}
 		_, out, cmdErr := runDriftCtlCmd(driftctlCmd)
 		if len(check.Env) > 0 {

